File: python3/maximum_gap.py
# Space : O(1)
# Time  : O(n log n)

class Solution:
    def maximumGap(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)

        if n <= 1:
            return 0

        nums.sort()

        for i in range(1, n):
            ans = max(ans, nums[i] - nums[i-1])

        return ans

File: golang/maximum_gap.go
package main

import "sort"

func maximumGap(nums []int) int {
	ans, n := 0, len(nums)

	if n <= 1 {
		return 0
	}

	sort.Ints(nums)
	for i := 1; i < n; i++ {
		ans = Max(ans, nums[i]-nums[i-1])
	}

	return ans
}
